src/test/kotlin/uk/gov/dluhc/logging/config/CorrelationIdMdcMessageListenerAspectTest.kt

```diff
@@ -0,0 +1,68 @@
+package uk.gov.dluhc.logging.config
+
+import ch.qos.logback.classic.Level
+import io.awspring.cloud.messaging.core.QueueMessagingTemplate
+import org.awaitility.kotlin.await
+import org.junit.jupiter.api.BeforeEach
+import org.junit.jupiter.api.Test
+import org.slf4j.MDC
+import org.springframework.beans.factory.annotation.Autowired
+import uk.gov.dluhc.logging.testsupport.assertj.assertions.ILoggingEventAssert
+import uk.gov.dluhc.printapi.config.IntegrationTest
```

Comment:
Hmmm, I had a feeling this would happen .... the `logging` package now has a dependency on the `printapi` package .... 🤔 

Even though we said we'd not want to do this, I wonder if there might be a case for 2 spring boot test contexts? (especially as its short'ish lived, and will one day move into the shared artefact repo)

What if there were an IntegrationTest superclass within the logging package somewhere that just brought in the beans we care about here ?

And perhaps we copy and paste the other dependencies (test log appender, and messageQueue, and anything else the other tests need ??)

We will have to do this anyway as and when we move it into the shared repo?

It might start identifying other elements that could be shared artefacts .... such as some of the test support stuff like the test log appender etc .... 🤔

Reply:
> What if there were an IntegrationTest superclass within the logging package somewhere that just brought in the beans we care about here ?

> And perhaps we copy and paste the other dependencies (test log appender, and messageQueue, and anything else the other tests need ??)

I agree @NathanRussellValtech.. Makes sense to me to do it as part of this work.

Reply:
Problem is that the test context will be based on the dependencies defined in the Gradle build file and when we move the code to a logging library the tests need to be re-written.  An example logging library module has been aded to `eip-ero-portal` but its `IntegrationTest` does not copy into this branch (as was hoped) due to the dependencies and Spring Boot auto-configuration.